src/connectors/assetpanda.py

```diff
@@ -0,0 +1,259 @@
+    ('BRAND', 'VARCHAR(256)'),
+    ('ASSIGNED_TO', 'VARIANT'),
+    ('SHARE_URL', 'VARCHAR(256)'),
+    ('OBJECT_VERSION_IDS', 'NUMBER(38,0)'),
+    ('CREATION_DATE', 'DATE'),
+    ('CREATED_BY', 'VARCHAR(256)'),
+    ('PURCHASE_PRICE', 'FLOAT'),
+    ('NEXT_SERVICE', 'DATE'),
+    ('BUILDING', 'VARIANT'),
+    ('CATEGORY', 'VARIANT'),
+    ('DESCRIPTION', 'VARCHAR(256)'),
+    ('CHANGED_BY', 'VARCHAR(256)'),
+    ('WIRELESS_STATUS', 'VARCHAR(256)'),
+    ('CREATED_AT', 'TIMESTAMP_TZ(9)'),
+    ('GPS_COORDINATES', 'VARIANT'),
+    ('UPDATED_AT', 'TIMESTAMP_TZ(9)'),
+    ('LOANER_POOL', 'BOOLEAN'),
+    ('DEFAULT_ATTACHMENT', 'VARIANT'),
+    ('ROOM', 'VARIANT'),
+    ('NOTES', 'VARCHAR(256)'),
+    ('OBJECT_DEPRECIATION', 'BOOLEAN'),
+    ('IS_EDITABLE', 'BOOLEAN'),
+    ('WIFI_MAC_ADDRESS', 'VARCHAR(256)'),
+    ('CHANGE_DATE', 'DATE'),
+    ('DISPLAY_SIZE', 'VARCHAR(256)'),
+    ('OPERATING_SYSTEM', 'VARCHAR(256)'),
+    ('SERIAL', 'VARCHAR(256)'),
+    ('END_OF_LIFE_DATE', 'DATE'),
+    ('IMEI_MEID', 'VARCHAR(256)'),
+    ('MODEL', 'VARCHAR(256)'),
+    ('MAC_ADDRESS', 'VARCHAR(256)'),
+    ('ENTITY', 'VARIANT'),
+    ('PO', 'VARCHAR(256)'),
+]
+
+
+def validate_key(item: dict, validate_dict: dict) -> None:
+    pass
+
+def validate_secret(secrets: dict) -> None:
+    pass
+
+def get_data(token: str, url: str, params: dict = {}) -> dict:
+    """
+    Make an API call
+    """
+    headers: dict = {
+        "Authorization": f"Bearer {token}"
+    }
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        req = requests.get(url, params=params, headers=headers)
+        print("req: ", req)
+        req.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise http_err
+    log.debug(req.status_code)
+
+    # decide whether to clean the data coming in
+
+    return req.json()
+    
+    
+def connect(connection_name, options):
+    """
+    Create the ingestion tables
+    """
+    print("connection_name: ", connection_name)
+    landing_table = f'data.assetpanda_{connection_name}_connection' # creates table in snowalert
+    
+    # comment = yaml_dump(module='assetpanda', **options)
+
+    db.create_table(name=landing_table, cols=LANDING_TABLE_COLUMNS, comment="AssetPanda Connector") # comment=comment)
+    db.execute(f'GRANT INSERT, SELECT ON {landing_table} TO ROLE {SA_ROLE}')
+
+    return {
+        'newStage': 'finalized',
+        'newMessage': "AssetPanda ingestion tables created!",
+    }
+
+def ingest(table_name, options):
+    landing_table = f'data.{table_name}'
+    timestamp = datetime.utcnow()
+    
+    asset_entity_id = options['asset_entity_id']
+    username = options['username']
+    secret = options['secret']
+    OAuth_client_id = options['OAuth_client_id']
+    OAuth_client_secret = options['OAuth_client_secret']
+
+    options['token'] = secret
```

Comment:
There is no 'token' defined in the 'CONNECTION_OPTIONS', so this would be invalid, also seems unnecessary.